pkg/platform/platform.go

```diff
@@ -32,12 +32,12 @@ import (
 var gBuilder builder.Builder
 
 // GetPlatformBuilder --
-func GetPlatformBuilder(ctx context.Context, c client.Client, namespace string) (builder.Builder, error) {
+func GetPlatformBuilder(c client.Client, namespace string) (builder.Builder, error) {
 	if gBuilder != nil {
 		return gBuilder, nil
 	}
 
-	gBuilder = builder.New(ctx, c, namespace)
+	gBuilder = builder.New(context.TODO(), c, namespace)
```

Comment:
Why creating a temp one and not propagating the existing one? You can pass a context.TODO if a context is not available at caller site.

Reply:
The problem is that the builder is created once per platform so if we use the context of the caller, it will be the one of the first caller and it can be closed which causes the builder to stop.

I need to think about a better solution but in a separate pr